packages/osd-pm/src/utils/validate_dependencies.ts

```diff
@@ -40,8 +43,43 @@ import { log } from './log';
 import { Project } from './project';
 import { ITree, treeToString } from './projects_tree';
 
-export async function validateDependencies(osd: OpenSearchDashboards, yarnLock: YarnLock) {
-  // look through all of the packages in the yarn.lock file to see if
+enum SingleVersionResolution {
+  STRICT = 'strict',
+  LOOSE = 'loose',
+  FORCE = 'force',
+  BRUTE_FORCE = 'brute-force',
+  IGNORE = 'ignore',
+}
+
+export async function validateDependencies(
+  osd: OpenSearchDashboards,
+  yarnLock: YarnLock,
+  /* `singleVersionResolution` controls how violations of single-version-dependencies is applied.
+   *    STRICT (default): throw an error and exit
+   *    LOOSE: identify and install a single version that satisfies all ranges
+   *    BRUTE_FORCE: identify and install the newest version
+   *    IGNORE: show all errors without exiting
+   *
+   * `LOOSE`:
+   *          Reconciles the various versions installed as a result of having multiple ranges for a dependency, by
+   *          choosing one that satisfies all said ranges. Even though installing the chosen version updates the
+   *          lock-files, no package.json changes would be needed.
+   *
+   * `BRUTE_FORCE`:
+   *          With no care for reconciliation, the newest of the various versions installed is chosen, irrespective of
+   *          whether it satisfies any of the ranges. Installing the chosen version updates the lock-files and a range
+   *          in the form of `^<version>` is applied to all `package.json` files that declared the dependency.
+   *
+   * `FORCE`:
+   *          For each dependency, first LOOSE resolution is attempted but if that fails, BRUTE_FORCE is applied.
+   *
+   * `IGNORE`:
```

Comment:
Any benefit by using this option more than get error as warning.  could we still run osd by ignore the error/warning. If not, should we callout specific dev scenario to leverage this option

Reply:
You can still run OSD with `IGNORE` and it will more than likely work just fine.